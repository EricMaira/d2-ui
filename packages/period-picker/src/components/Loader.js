import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

const styles = () => {
    return {
        loader: {
            padding: 16,
            textAlign: 'center',
        },
    };
};

const Loader = ({ classes }) => {
    return (
        <div className={classes.loader}>
            <CircularProgress size={29.5} thickness={2.5} />
        </div>
    );
};

Loader.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Loader);
