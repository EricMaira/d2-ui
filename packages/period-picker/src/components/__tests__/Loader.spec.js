import React from 'react';
import { shallow } from 'enzyme';
import { Loader } from '../Loader';

describe('<Loader/>', () => {
    it('Matches the snapshot', () => {
        const props = { classes: { loader: 'loader' } };
        const wrapper = shallow(<Loader {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
