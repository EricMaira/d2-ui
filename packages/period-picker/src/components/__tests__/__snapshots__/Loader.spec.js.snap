// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> Matches the snapshot 1`] = `
<div
  className="loader"
>
  <WithStyles(CircularProgress)
    size={29.5}
    thickness={2.5}
  />
</div>
`;
